src/sbvr-api/sbvr-utils.coffee

```diff
@@ -659,15 +662,25 @@ exports.handleODataRequest = handleODataRequest = (req, res, next) ->
 						catch err
 							api[apiRoot].logger.error('Failed to compile abstract sql: ', request.abstractSqlQuery, err, err.stack)
 							throw new SqlCompilationError(err)
+						request.modifiedFields = AbstractSQLCompiler.getModifiedFields(request.abstractSqlQuery)
 					return request
 
 			.then (request) ->
 				# Run the request in its own transaction
 				runTransaction req, request, (tx) ->
 					if _.isArray request
-						env = new Map()
+						env =
+							results: new Map()
+							validations: new Map()
 						Promise.reduce(request, runChangeSet(req, res, tx), env)
-						.then (env) -> Array.from(env.values())
+						.tap (env) ->
+							validations = Array.from(env.validations.values())
+							Promise.map validation, (requestToValidate) ->
```

Comment:
I think this whole PR can be made simpler and more performant by just adding on to the modified fields, so you end up with all of the tables and fields that the changeset modfies and then run the validation once which will still run all potentially affected rules and also guarantee that any given rule will only be run once per changeset